Add a summary of our general Tech Infrastructure (#93)
srobo/runbook: docs/infrastructure/README.md

# Overview

> Student Robotics has a variety of infrastrcture in order to support its mission.

Reviewer: ```suggestion
Student Robotics has a variety of infrastructure in order to support its goals.
```

Author: Typo fixed in 24fe613712cbad13a58da8fedcec70ab48797726. Is there a reason to change mission to goals?

This document aims to provide an outline of the general structure.

Currently our infrastructure as a whole does not have a designed owner, instead
individual pieces are looked after by volunteer Teams or just by volunteers who

> have historically maintained them.

Reviewer: Might be worth referencing the Infrastructure Team proposal here, or waiting until that proposal is finalised.

Author: I'd honestly hoped that that proposal would be sorted by now. Until it does appear I'm mixed on whether mentioning it is a good idea. If it doesn't appear soon, then over time it would become much less useful for this to mention it.

Author: 1026b4d acknowledges the infra team, however I'm deliberately omitting a description of who's responsible for what for now. I think we'll want to work out a bit more generally where in the runbook to put "team X is responsible for Y" type information as right now there isn't one obvious home for it.

## Website

Our website consists of a main public facing site at https://studentrobotics.org
as well as a collection of other sites which appear at various paths off the
root domain. This runbook is an example of such a sub-site, as are the
competitor-facing documentation which appears at [`/docs`][slash-docs].

> These sub sites (and the main website itself) are hosted as GitHub Pages, with a

Reviewer 2: Not all the sites are GitHub pages, ie the IDE.

If the idea is to emphasise that things like the docs and comp site might _look_ like they're 1 site, but are actually separate, then that's probably worth explaining separately.

Author: I'm intending to use the term "sub site" here to mean the things which are _websites_ (i.e: things with pages) rather than web apps or other services. I'd also argue that such things are not part of our _website_.
I've attempted to hint at this in the next sentence by also mentioning "other services". I'd also hope that it was clear (from the subsequent sections) that we have plenty of things which aren't hosted on GitHub Pages.

Author: ce75d2f3f7caf3cacb6d0df46e874ab8c7f668a6

separate NGINX instance acting as a [reverse-proxy][reverse-proxy] to facade
these and other services onto the root domain. Currently we use Digital Ocean
for the hosting of our [infrastructure][infrastructure] as well as for DNS and

> TLS termination. Our domains are registered with Namecheap.

Reviewer: ```suggestion
TLS termination. Our domains are registered with Namecheap and are managed by the Trustees.
```

Author: This is not strictly true -- the DNS entries for them are in Digital Ocean (as this paragraph notes) and the actual technical management of the domains is therefore a bit grey in terms of who manages it. I've deliberately opted to leave that detail out for now. Once the Infrastructure Team exists, this will naturally become clearer (and an easy fix to document).

[slash-docs]: https://studentrobotics.org/docs
[reverse-proxy]: https://github.com/srobo/reverse-proxy
[infrastructure]: https://github.com/srobo/infrastructure

### Competitor Services

During the competition, there are a number of [hosted services](./competitor-services.md)

> which the competitors use. These have thier own managed configuration and are

Reviewer: ```suggestion
which the competitors use. These have their own managed configuration and are
```

> typically hosted for the duration of the competition "year" only.

Reviewer: ```suggestion
typically hosted for the duration of the *competition year* only.
```

Competition year is defined in the glossary, so can be used with italics here.

### Competition Services

During the competition event we use the [SRComp suite][srcomp-suite] of tools to
help run the matches, including a VM to host a public-facing copy of the HTTP
API for use on the website.

[srcomp-suite]: https://github.com/PeterJCLaw/srcomp/wiki

## Development: GitHub

A lot of our collaborative work (including hardware and software development, as
well as general task management) takes place [on GitHub][git-and-github], within our
[`srobo`](https://github.com/srobo) organisation.

We also have a separate [`srobo-legacy`](https://github.com/srobo-legacy)
organisation which contains some archived git repositories that pre-date our use
of GitHub. Moving these over to the main organisation and de-duplicating the
repositories is an [ongoing project][legacy-repository-migration].

[git-and-github]: ../volunteering/git-and-github.md
[legacy-repository-migration]: https://github.com/srobo/tasks/issues/179

## Email, Documents: Google

For our email (`@studentrobotics.org`) accounts as well as document sharing we

> use Google's G-Suite. Google Groups are used for various mailing lists.

Reviewer: ```suggestion
use Google Workspace. Google Groups are used for various mailing lists.
```

G Suite is now Google Workspace.

Might be worth differentiating between the `srobo-*@googlegroups` and `@studentrobotics` mailing lists here.

## External mailing lists: Mailchimp

> For sending some batch emails we use Mailchimp.

Reviewer: This section is a bit empty

- Link to MailChimp
- Link to repos where emails are drafted
- List of mailing lists used here?

Author: Indeed. This is a bit of a difficult one because for example the [team-emails](https://github.com/srobo/team-emails/) repo really is part of the Competition Team rather than general SR infrastructure. The [volunteer-emails](https://github.com/srobo/volunteer-emails/) (and maybe [srawn](https://github.com/srobo/srawn)) repos are a bit more ambiguous.

I'm happy to link to MailChimp, though without suitable login credentials (which obviously cannot be included here) it's rather moot.

I'll add these for now, though when https://github.com/srobo/runbook/issues/71 is addressed it should revisit what fits here.

Author: I've attempted to address this in 247fd4e3923ce270a94ff6e51f74576dcabafd46. Do you think that covers it? (SR(A)WN felt enough outside our Infrastructure that I've raised https://github.com/srobo/runbook/issues/94 rather than shoe-horn it into this PR).

Author: I'm going to resolve this thread, though it would be great to link to the SR(A)WN page once #97 merges.
